### DISCUSSION TITLE
Adding a dashed line for the average with a label for the exact value on the y-axis

### DISCUSSION THREAD

**Discussion by MattyKems87** [recharts/recharts#3047] [Q&A] [answered]
Hi All

For a couple of our graphs, we want to display the average value with a dashed line, as shown in the following image:
![image](https://user-images.githubusercontent.com/80238118/200209500-bc4de56a-9dc9-43e3-8f4f-0addcc970532.png)

We would also like to display a label on the y-axis that shows the exact average value. Is there a way we could implement this using the API? 

Thank you

**Comment by ckifer** [comment, marked as answer]
You can use a `ReferenceLine` for the dashed line. You can use a label on the reference line for the label, it wouldn't be exactly on the Y axis but you could get it close.

https://codesandbox.io/p/sandbox/line-chart-ref-line-label-3w2kkx

**Reply by teukuamru** [reply]
> You can use a `ReferenceLine` for the dashed line. You can use a label on the reference line for the label, it wouldn't be exactly on the Y axis but you could get it close.
> 
> https://codesandbox.io/s/line-chart-width-reference-line-forked-7c9des?file=/src/App.tsx

Hi! The link that you shared is not accessible anymore. Can you please provide the code here?

**Reply by ckifer** [reply]
Updated sandbox https://codesandbox.io/p/sandbox/line-chart-ref-line-label-3w2kkx

**Reply by ckifer** [reply]
Code

```tsx
      <LineChart data={data} margin={{ left: 50 }}>
        <XAxis dataKey="name" />
        <YAxis />
        <Tooltip />
        <Legend />
        <ReferenceLine y={2000} strokeDasharray="10 10" stroke="#111">
          <Label value={2000} fill="#111" position="left" offset={10} />
        </ReferenceLine>
        <Line
          type="monotone"
          dataKey="pv"
          stroke="#8884d8"
          activeDot={{ r: 8 }}
        />
        <Line type="monotone" dataKey="uv" stroke="#82ca9d" />
      </LineChart>
```
